tests/test_declaration.py

```diff
@@ -657,3 +689,53 @@ def test_dict(self):
 
         with self.assertRaises(KeyError):
             del node.style['no-such-property']
+
+    def test_font_shorthand_property(self):
+        node = TestNode(style=CSS())
+        node.layout.dirty = None
+
+        # Check initial value
+        self.assertEqual(node.style.font, 'normal normal normal medium/normal initial')
+
+        # Check Initial values
+        self.assertEqual(node.style.font_style, 'normal')
+        self.assertEqual(node.style.font_weight, 'normal')
+        self.assertEqual(node.style.font_variant, 'normal')
+        self.assertEqual(node.style.font_size, 'medium')
+        self.assertEqual(node.style.line_height, 'normal')
+        self.assertEqual(node.style.font_family, ['initial'])
+
+        # Check individual properties update the unset shorthand
+        node.style.font_style = 'italic'
+        node.style.font_weight = 'bold'
+        node.style.font_variant = 'small-caps'
+        node.style.font_size = '10px'
+        node.style.line_height = '1.5'
+        node.style.font_family = ['"DejaVu Sans"', 'serif']
+        # TODO: Is this the behavior we want?
+        self.assertEqual(node.style.font, 'italic small-caps bold 10.0px/1.5 "DejaVu Sans", serif')
+
+        # Check setting the shorthand resets values
+        node.style.font = '9px serif'
+        self.assertEqual(node.style.font_style, 'normal')
+        self.assertEqual(node.style.font_weight, 'normal')
+        self.assertEqual(node.style.font_variant, 'normal')
+        self.assertEqual(node.style.line_height, 'normal')
+        self.assertEqual(str(node.style.font_size), '9.0px')
+        self.assertEqual(node.style.font_family, ['serif'])
+        self.assertEqual(node.style.font, '9px serif')
+
+        # Check individual properties do not update the set shorthand
+        # TODO: Is this the behavior we want?
+        node.style.font = '9px serif'
+        node.style.font_style = 'italic'
+        node.style.font_weight = 'bold'
+        node.style.font_variant = 'small-caps'
+        node.style.font_size = '10px'
+        node.style.line_height = '1.5'
+        node.style.font_family = ['"DejaVu Sans"', 'serif']
+        self.assertEqual(node.style.font, '9px serif')
```

Comment:
This isn't the behavior I'd expect (especially if they're set in this order) - I'd expect the final font str to be `italic small-caps bold 10px/1.5 "DejaVu Sans", serif`. This matches what you'd expect if the calls were made sequentially (and also matches CSS behavior if the declarations are ordered in the same order)

Reply:
Got it, was not sure of the behavior and did not test it myself